grpc-context-utils/src/main/java/org/hypertrace/core/grpcutils/context/RequestContextConstants.java

```diff
@@ -17,10 +17,20 @@
 
   public static final String AUTHORIZATION_HEADER = "authorization";
 
+  /** The values in this set are looked up with case insensitivity. */
+  public static final Set<String> HEADER_PREFIXES_TO_BE_PROPAGATED =
+      Set.of(
+          TENANT_ID_HEADER_KEY,
+          "X-B3-",
+          "grpc-trace-bin",
+          "traceparent",
+          "tracestate",
+          AUTHORIZATION_HEADER);
+
   /**
-   * The values in this set are looked up with case insensitivity.
+   * These headers may affect returned results and should be accounted for in any cached remote
+   * results
    */
-  public static final Set<String> HEADER_PREFIXES_TO_BE_PROPAGATED =
-      Set.of(TENANT_ID_HEADER_KEY, "X-B3-", "grpc-trace-bin",
-          "traceparent", "tracestate", AUTHORIZATION_HEADER);
+  static final Set<String> CACHE_MEANINGFUL_HEADERS =
```

Comment:
I know there's some debate whether authorization is meaningful. I'll remove it if needed to unblock this since we can revisit this later in a common way, but I maintain it's better to have the default implementation err on the side of being always correct at the cost of potential duplication.